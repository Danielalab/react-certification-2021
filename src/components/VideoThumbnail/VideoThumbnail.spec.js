import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import VideoThumbnail from './VideoThumbnail.component';
import dataDummy from '../../hooks/tests/responses/searchQuerySuccess.json';

describe('VideoThumbnail', () => {
  let container;

  beforeEach(() => {
    const { id, snippet } = dataDummy.items[0];
    container = render(<VideoThumbnail id={id.videoId} videoData={snippet} />, {
      wrapper: MemoryRouter,
    }).container;
  });

  test('Should renders VideoThumbnail component', () => {
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should contains an image with the thumbnail url', () => {
    const videoThumbnailImageElement = screen.getByAltText(
      'fake video title 1 thumbnail'
    );
    expect(videoThumbnailImageElement).toBeInTheDocument();
    expect(videoThumbnailImageElement).toHaveAttribute('src', 'fake-video-url-image.png');
  });

  test('Should contains an image with the channel avatar', () => {
    const channelImageElement = screen.getAllByAltText('fake channel title')[0];
    expect(channelImageElement).toBeInTheDocument();
    expect(channelImageElement).toHaveAttribute('src', 'undraw_default_avatar.svg');
  });

  test('Should contains a link redirects to the video url', () => {
    const linkELement = screen.getByText(/fake video title 1/i);
    expect(linkELement).toBeInTheDocument();
    expect(linkELement).toHaveAttribute('href', '/fakeVideoId1');
  });

  test('Should shows the channel title', () => {
    const channelTitleElement = screen.getAllByText(/fake channel title/i)[0];
    expect(channelTitleElement).toBeInTheDocument();
  });

  test('Should shows the video published date', () => {
    const channelTitleElement = screen.getAllByText(/2014-09-27T01:39:18Z/i)[0];
    expect(channelTitleElement).toBeInTheDocument();
  });
});
