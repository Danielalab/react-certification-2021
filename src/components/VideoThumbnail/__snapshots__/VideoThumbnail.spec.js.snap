// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoThumbnail Should renders VideoThumbnail component 1`] = `
<div>
  <div
    class="sc-bdnxRM iIRjbH"
  >
    <figure>
      <img
        alt="fake video title 1 thumbnail"
        src="fake-video-url-image.png"
      />
    </figure>
    <div
      class="sc-gtsrHT eMBaCB"
    >
      <img
        alt="fake channel title"
        class="sc-hKFxyN bdMQIT"
        src="undraw_default_avatar.svg"
      />
      <div
        class="sc-jSFjdj bgycdW margin-x"
      >
        <a
          class="sc-dlnjwi hzuETT"
          href="/fakeVideoId1"
        >
          fake video title 1
        </a>
        <div>
          <p
            class="sc-eCApnc fpRBdY"
          >
            fake channel title
          </p>
          <p
            class="sc-eCApnc fpRBdY"
          >
            2014-09-27T01:39:18Z
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
